core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2209,6 +2209,14 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
         if (!doubleSpendTxns.isEmpty()) {
             // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
             killTxns(doubleSpendTxns, tx);
+
+            // disconnect irrelevant inputs (otherwise might cause serialization issue)
+            for(TransactionInput input : tx.getInputs()) {
```

Comment:
Well, it is only disconnecting the irrelevant outputs, although I'm not sure about the implication of that.
I've also checked that it can disconnect an irrelevant link between 2 relevant transactions.